tests/suite/test_prometheus_endpoint.py

```diff
@@ -0,0 +1,148 @@
+    else:
+        kube_apis.apps_v1_api.patch_namespaced_daemon_set(ingress_controller, namespace, body)
+    wait_until_all_pods_are_ready(kube_apis.v1, namespace)
+
+
+@pytest.fixture(scope="class")
+def ingress_setup(request, kube_apis, ingress_controller_endpoint, test_namespace) -> IngressSetup:
+    print("------------------------- Deploy Ingress Example -----------------------------------")
+    secret_name = create_secret_from_yaml(
+        kube_apis.v1, test_namespace, f"{TEST_DATA}/smoke/smoke-secret.yaml"
+    )
+    create_items_from_yaml(
+        kube_apis, f"{TEST_DATA}/smoke/standard/smoke-ingress.yaml", test_namespace
+    )
+    ingress_host = get_first_ingress_host_from_yaml(
+        f"{TEST_DATA}/smoke/standard/smoke-ingress.yaml"
+    )
+    create_example_app(kube_apis, "simple", test_namespace)
+    wait_until_all_pods_are_ready(kube_apis.v1, test_namespace)
+    ensure_connection_to_public_endpoint(
+        ingress_controller_endpoint.public_ip,
+        ingress_controller_endpoint.port,
+        ingress_controller_endpoint.port_ssl,
+    )
+
+    def fin():
+        print("Clean up simple app")
+        delete_common_app(kube_apis, "simple", test_namespace)
+        delete_items_from_yaml(
+            kube_apis, f"{TEST_DATA}/smoke/standard/smoke-ingress.yaml", test_namespace
+        )
+        delete_secret(kube_apis.v1, secret_name, test_namespace)
+
+    request.addfinalizer(fin)
+
+    return IngressSetup(ingress_host)
+
+
+@pytest.mark.demo
+class TestPrometheusEndpoint:
+    @pytest.mark.parametrize(
+        "ingress_controller, expected_metrics",
+        [
+            pytest.param(
+                {"extra_args": ["-enable-prometheus-metrics", "-enable-latency-metrics", "-prometheus-tls-secret=nginx-ingress/prometheus-test-secret"]},
+                [
+                    'nginx_ingress_controller_ingress_resources_total{class="nginx",type="master"} 0',
+                    'nginx_ingress_controller_ingress_resources_total{class="nginx",type="minion"} 0',
+                ],
+            )
+        ],
+        indirect=["ingress_controller"],
+    )
+    def test_https_metrics(
+            self,
+            prometheus_setup,
+            ingress_controller_endpoint,
+            ingress_controller,
+            enable_exporter_port,
+            expected_metrics,
+            ingress_setup,
+    ):
+        resp = {}
+
+        # assert http fails
+        req_url = f"http://{ingress_controller_endpoint.public_ip}:{ingress_controller_endpoint.metrics_port}/metrics"
+        try:
+            resp = requests.get(req_url, verify=False)
+        except:
+            print("request fails as expected")
+
+        assert resp.status_code == 400, f"Expected 400 code for http request to /metrics but got {resp.status_code}"
+
+        # assert https succeeds
+        req_url = f"https://{ingress_controller_endpoint.public_ip}:{ingress_controller_endpoint.metrics_port}/metrics"
```

Comment:
will not work for both NodePort and Loadbalancer service?

Reply:
I'm not sure what your comment means, can you rephrase it?

Reply:
typo. "will it work ... " , my bad!

I was concerned whether accessing the Prometheus endpoint will work in the test when the IC is exposed through a service of the type LoadBalancer.

This answers my question https://github.com/nginxinc/kubernetes-ingress/blob/79c98c311223fec1c1d956e17e40dbc58f0d68da/tests/suite/fixtures.py#L129 -- port 9113 is exposed through a service and it will work whether the service type is LoadBalancer and NodePort.